In [1]:
from sampler import do_sample, TimedMoviepyAMIGOSSampler, TimedFastAMIGOSSampler

In [2]:
%%time
%%capture --no-display
# ~ 4 minutes for my current subset of data.
path = "./../../resources/AMIGOS/"
sampler = TimedFastAMIGOSSampler(path + "sampled/", 5)
do_sample(path, path + "sampled/", sampler, 100, True)

File './../../resources/AMIGOS/sampled/P01_5/0.mp4' already exists. Overwrite? [y/N] File './../../resources/AMIGOS/sampled/P01_5/1.mp4' already exists. Overwrite? [y/N] File './../../resources/AMIGOS/sampled/P01_5/2.mp4' already exists. Overwrite? [y/N] File './../../resources/AMIGOS/sampled/P01_5/3.mp4' already exists. Overwrite? [y/N] File './../../resources/AMIGOS/sampled/P01_5/4.mp4' already exists. Overwrite? [y/N] File './../../resources/AMIGOS/sampled/P01_5/5.mp4' already exists. Overwrite? [y/N] File './../../resources/AMIGOS/sampled/P01_5/6.mp4' already exists. Overwrite? [y/N] File './../../resources/AMIGOS/sampled/P01_5/8.mp4' already exists. Overwrite? [y/N] File './../../resources/AMIGOS/sampled/P01_5/7.mp4' already exists. Overwrite? [y/N] File './../../resources/AMIGOS/sampled/P01_5/9.mp4' already exists. Overwrite? [y/N] File './../../resources/AMIGOS/sampled/P01_5/10.mp4' already exists. Overwrite? [y/N] File './../../resources/AMIGOS/sampled/P01_5/11.mp4' already exi

CPU times: user 6.16 s, sys: 4.12 s, total: 10.3 s
Wall time: 4min 24s


#### With ProcessPoolExecutor
CPU times: user 6.16 s, sys: 4.12 s, total: 10.3 s <br>
Wall time: 4min 24s

#### With ThreadPoolExecutor
CPU times: user 6.08 s, sys: 2.99 s, total: 9.07 s <br>
Wall time: 4min 17s

In [ ]:
%%time
%%capture --no-display
path = "./../../resources/AMIGOS/"
sampler = TimedMoviepyAMIGOSSampler(path + "sampled/", 5)
do_sample(path, path + "sampled/", sampler, 100)

In [1]:
%%time
%%capture --no-display
# Does not work too well.
path = "./../../resources/AMIGOS/"
sampler = TimedMoviepyAMIGOSSampler(path + "sampled/", 5)
do_sample(path, path + "sampled/", sampler, 100, True)

CPU times: user 11min 20s, sys: 1min 44s, total: 13min 4s
Wall time: 15min 46s


In [ ]:
# CPU times: user 17.7 s, sys: 10.5 s, total: 28.1 s
# Wall time: 9min 54s